big_tests/tests/rest_SUITE.erl

```diff
@@ -172,11 +222,17 @@ user_can_be_registered_and_removed(_Config) ->
     {?OK, Lusers1} = gett(admin, <<"/users/localhost">>),
     assert_inlist(<<"mike@", Domain/binary>>, Lusers1),
     % try to create the same user
-    {?ERROR, _} = post(admin, <<"/users/localhost">>, CrUser),
+    {?FORBIDDEN, _} = post(admin, <<"/users/localhost">>, CrUser),
     % delete user
     {?NOCONTENT, _} = delete(admin, <<"/users/localhost/mike">>),
     {?OK, Lusers2} = gett(admin, <<"/users/localhost">>),
     assert_notinlist(<<"mike@", Domain/binary>>, Lusers2),
+    % invalid jid
+    CrBadUser = #{username => <<"m@ke">>, password => <<"nicniema">>},
+    {?BAD_REQUEST, <<"Invalid jid", _/binary>>} = post(admin, <<"/users/localhost">>, CrBadUser),
+    {?BAD_REQUEST, <<"Invalid jid", _/binary>>} = delete(admin, <<"/users/localhost/@mike">>),
+%%    {?FORBIDDEN, _} = delete(admin, <<"/users/localhost/mike">>), % he's already gone, but we
```

Comment:
i think we must report an error here, otherwise it's hard to recognize the case when we have a typo in a user name

Reply:
also it leaves untested the [implementation](https://github.com/esl/MongooseIM/pull/2908/files#diff-62e4b0dd41ca87f36493646142ca949a66970e8175bd6ddb936b5e556735575fR277)

Reply:
There is an explanation right below this line - this is how ejabberd_auth works. Unless we change that one, maybe in a later stage.

Reply:
let's mock it

Reply:
There is a terrible mess in auth backends. Let's leave it as it is for now, and next thing I do is clean it up, so that they all return what they should.

Reply:
Though maybe doing this within this PR would be more consistent, I don't know.

Reply:
This PR is big enough, don't add here auth backend rework.
If you want to add this check in one of the following PRs then I'm fine with it.
just ensure that we won't forget about it, m.b. add notes in the corresponding jira issue